### PR TITLE
Add proxy support

### PR DESCRIPTION
**Pull request by robertfoss**
Response to issue to #428.

This feature utilises the proxy support of the requests module, and
has been tested with SOCKS5 through a SSH tunnel, and has been
verified to be working.

Consider this a RFC, any polish is most welcome.

Try it out using:
```
$ ssh -D 18888 your.domain.here
$ svtplay-dl --proxy=socks5://localhost:18888 "https://www.svtplay.se/video/15722552/winnie/dox-winnie-avsnitt-1"
```

### CODE REVIEW

File: lib/svtplay_dl/__init__.py [hunk 1]
```diff
@@ -442,7 +442,9 @@ def main():
     parser.add_option("--cmore-operatorlist", dest="cmoreoperatorlist", default=False, action="store_true",
                       help="show operatorlist for cmore")
     parser.add_option("--cmore-operator", dest="cmoreoperator", default=None, metavar="operator")
-                      
+    parser.add_option("--proxy", dest="proxy", default=None,
+                      metavar="proxy", help="access network through http/socks proxy")
```
Comment (dentarg) [inline lib/svtplay_dl/__init__.py]: I think an example in the help text would be good, so you understand that you need the scheme for instance.

Here's how youtube-dl phrases it: https://github.com/rg3/youtube-dl/blob/master/README.md#network-options

The requests docs about proxies might also provide some inspiration: http://docs.python-requests.org/en/master/user/advanced/#proxies

File: lib/svtplay_dl/__init__.py [hunk 2]
```diff
@@ -468,6 +470,11 @@ def main():
         c.operatorlist()
         sys.exit(0)
 
+    if options.proxy:
+        options.proxy = options.proxy.replace("socks5", "socks5h", 1)
```
Comment (dentarg) [inline lib/svtplay_dl/__init__.py]: TIL [`socks5` vs `socks5h`](https://github.com/shazow/urllib3/pull/1036) 🤓

**Comment by robertfoss** [comment]
This patch also introduces a dependency on requests[socks], where should it be documented?

**Comment by robertfoss** [comment]
@spaam do you have any input?

**Comment by dentarg** [comment]
> This patch also introduces a dependency on requests[socks], where should it be documented?

You could add it to the list in this section https://github.com/spaam/svtplay-dl#from-source I guess

> I just added it to setup.py

I see that [PySocks](https://github.com/Anorov/PySocks) was added and not [requests[socks]](http://docs.python-requests.org/en/master/user/advanced/#socks), is that the same thing?

**Comment by robertfoss** [comment]
As far as I understand [PySocks](https://github.com/Anorov/PySocks) is [requests[socks]](http://docs.python-requests.org/en/master/user/advanced/#socks).

File: README.rst [hunk 3]
```diff
@@ -80,6 +80,7 @@ you don’t have them, some features will not be working.
 -  `RTMPDump`_ 2.4 or higher to download RTMP streams.
 -  `PyCrypto`_ to download encrypted HLS streams
 -  `Requests`_
+-  `PySocks`_ to enable proxy support
```
Comment (dentarg) [inline README.rst]: You want to add a link (see the bottom of the document) for this to be rendered correctly

Comment (robertfoss) [reply README.rst]: Oups! Fixd!

**Comment by spaam** [comment]
oh fancy

**Pull request merged**